worker/sample_test.go

```diff
@@ -0,0 +1,238 @@
+package main
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+)
+
+func TestSampleSuccess(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
As with the main tests (but more urgent because of the multiple usages here) I would set this httptest server up once https://golang.org/pkg/testing/#hdr-Main

Reply:
cool